core/internal/runhistoryreader/reader_test.go

```diff
@@ -1,119 +1,240 @@
 package runhistoryreader
 
 import (
-	"bytes"
 	"encoding/json"
 	"fmt"
-	"io"
+	"math"
 	"net/http"
+	"net/http/httptest"
 	"os"
 	"path/filepath"
 	"testing"
 
+	"github.com/apache/arrow-go/v18/arrow"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"github.com/wandb/wandb/core/internal/gqlmock"
+	"github.com/wandb/wandb/core/internal/runhistoryreader/parquet/iterator/iteratortest"
 )
 
-// mockRoundTripper is a mock implementation of http.RoundTripper for testing
-type mockRoundTripper struct {
-	requestURL     string
-	responseBody   []byte
-	responseStatus int
-	capturedURL    string
-}
+func respondWithParquetContent(
+	t *testing.T,
+	parquetContent []byte,
+) func(responseWriter http.ResponseWriter, request *http.Request) {
+	t.Helper()
 
-func (m *mockRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
-	// Capture the request URL for verification
-	m.capturedURL = req.URL.String()
+	return func(responseWriter http.ResponseWriter, request *http.Request) {
+		responseWriter.Header().Set("Accept-Ranges", "bytes")
 
-	// Return mock response
-	return &http.Response{
-		StatusCode: m.responseStatus,
-		Body:       io.NopCloser(bytes.NewReader(m.responseBody)),
-		Header:     make(http.Header),
-	}, nil
+		// Handle range requests for remote reading
+		rangeHeader := request.Header.Get("Range")
+		if rangeHeader != "" {
+			var start, end int64
+			_, err := fmt.Sscanf(rangeHeader, "bytes=%d-%d", &start, &end)
+			require.NoError(t, err)
+			min := int64(math.Min(float64(end+1), float64(len(parquetContent))))
```

Comment:
Why convert to `float64` and then to `int64` instead of converting to `int64` directly?